lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

```diff
@@ -157,242 +217,1038 @@
+            results['port_binding'] = self.port_binding
+            pg_spec.type = self.get_api_port_binding_mode(self.port_binding)
+        # Port allocation
+        if not self.port_allocation:
+            if self.port_binding == 'dynamic' or self.port_binding == 'ephemeral':
+                results['port_allocation'] = 'n/a'
+            else:
+                results['port_allocation'] = 'elastic'
+                pg_spec.autoExpand = True
+        else:
+            results['port_allocation'] = self.port_allocation
+            if self.port_allocation == 'elastic':
+                pg_spec.autoExpand = True
+            elif self.port_allocation == 'fixed':
+                pg_spec.autoExpand = False
+
+        # Advanced / Port policies
+        results['adv_reset_at_disconnect'] = self.pp_reset
+        results['adv_block_ports'] = self.pp_block_ports
+        results['adv_traffic_shaping'] = self.pp_shaping
+        results['adv_vendor_conf'] = self.pp_vendor_conf
+        results['adv_vlan'] = self.pp_vlan
+        results['adv_uplink_teaming'] = self.pp_uplink_teaming
+        results['adv_network_rp'] = self.pp_network_rp
+        results['adv_security'] = self.pp_security
+        results['adv_netflow'] = self.pp_netflow
+        results['adv_traffic_filtering'] = self.pp_traffic_filter
+        result = self.check_port_policy_config(pg_config=None)
+        pg_spec.policy = result[0]
+
+        pg_spec.defaultPortConfig = vim.dvs.VmwareDistributedVirtualSwitch.VmwarePortConfigPolicy()
+
+        # Network resource pool
+        results['network_rp'] = self.network_resource_pool
+        if self.network_resource_pool == 'default':
+            pg_spec.defaultPortConfig.networkResourcePoolKey = vim.StringPolicy(inherited=False, value='-1')
+        else:
+            network_resource_pool = self.find_network_rp_by_name(self.network_resource_pool)
+            if network_resource_pool is None:
+                self.module.fail_json(msg="Network resource pool '%s' was not found" % self.network_resource_pool)
+            pg_spec.defaultPortConfig.networkResourcePoolKey = vim.StringPolicy(
+                inherited=False, value=network_resource_pool.key
+            )
+
+        # Security
+        results['sec_promiscuous_mode'] = self.sec_promiscuous_mode
+        results['sec_mac_changes'] = self.sec_mac_changes
+        results['sec_forged_transmits'] = self.sec_forged_transmits
+        result = self.check_security_config(pg_config=None)
+        pg_spec.defaultPortConfig.securityPolicy = result[0]
 
-        if not self.module.check_mode:
-            task = self.dvs_portgroup.Destroy_Task()
-            changed, result = wait_for_task(task)
-        self.module.exit_json(changed=changed, result=str(result))
+        # TODO: Traffic shaping
 
-    def state_exit_unchanged(self):
-        self.module.exit_json(changed=False)
+        if self.vlan_id == 0 and self.vlan_trunk_range is None and self.private_vlan_id is None:
```

Comment:
Because it already checks the exclusivity of these 3 options in the module definition, this check can be minimized to `if self.vlan_id == 0:` (the other 2 will automatically be None)

Reply:
Makes sense. Thx.